scripts/validate-schemas-final.js

```diff
@@ -0,0 +1,71 @@
+const fs = require('fs');
+const path = require('path');
+
+
+function validation (excludedFiles){
+
+  // Specify the path to the 'schemas' directory
+  const directoryPath = './schemas';
+
+
+  const files = fs.readdirSync(directoryPath);
+
+  // Filter files
+  const filteredFiles = files.filter(file => !excludedFiles.includes(file));
+
+
+  // Iterate through the filtered files
+  filteredFiles.forEach(file => {
+    // Construct the full path to the JSON schema file
+    const filePath = path.join(directoryPath, file);
+
+    try {
+      // Read and parse the JSON schema
+      const fileContent = fs.readFileSync(filePath, 'utf8');
+      const obj = JSON.parse(fileContent);
+
+        if(obj.$schema === 'http://json-schema.org/draft-04/schema'){
+
+            // console.log('draft-04'); // for debugging purpose
+            const Ajv = require('ajv-draft-04');
```

Comment:
Since this `require` is located inside the `forEach` loop, you are creating a new AJV instance on each iteration. This is not only not needed but not good in terms of performance and resource consumption.

I suggest you to move your requires to the top of the file. You can change the const name to `AjvDraft07` and `AjvDraft04` so you can either use one or another later here.

Reply:
> Since this require is located inside the forEach loop, you are creating a new AJV instance on each iteration. This is not only not needed but not good in terms of performance and resource consumption.

Hmm... didn't knew this, Thank you for teaching me this! :pray: 

> I suggest you to move your requires to the top of the file. You can change the const name to AjvDraft07 and AjvDraft04 (or any other name of your preference) so you can either use one or another later here.

Will implement your suggestion... :+1:

Reply:
Hey Sergio!
I've updated the script for the instantiation part, please see here: 3570fe0b2cdc0d47287fad04eeab40e09ba7150c

Reply:
Resolved with 3570fe0